src/pages/sign-in/index.tsx

```diff
@@ -2,27 +2,61 @@ import React, { useState } from "react"
 import Image from "next/image"
 import SignInTextField from "@/components/text-fields/sign-in-text-field"
 import { useRouter } from "next/router"
-import ErrorPopup from "@/components/popups/error-popup"
 import MainButton from "@/components/buttons/main-button"
 import { useSupabaseClient } from "@supabase/auth-helpers-react"
 import { Database } from "@/types/database"
 import { AuthResponse } from "@supabase/gotrue-js"
+import ErrorMsg from "@/components/error/error-msg"
+import { FormDataItem } from "@/pages/change-password"
+import { useForm } from "react-hook-form"
 
 const Index = () => {
-  const [email, setEmail] = useState<string>("")
-  const [password, setPassword] = useState<string>("")
-  const [showErrorPopup, setErrorPopup] = useState<boolean>()
+  const [showErrorMsg, setShowErrorMsg] = useState<boolean>()
   const [errorPopupText, setErrorPopupText] = useState<string>("")
   const supabase = useSupabaseClient<Database>()
-
   const router = useRouter()
+  const {
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm()
+
+  const email = watch("email")
+  const password = watch("password")
+
+  const signInFormData: FormDataItem[] = [
+    {
+      tag: "email",
+      type: "text",
+      placeholder: "Email",
+      dataCy: "sign-in-email",
+      validationOptions: {
+        required: "Email is required",
+        pattern: {
+          value: /\S+@\S+\.\S+/,
```

Comment:
##@##.com registers as a valid email with this pattern, and so will lots of other invalid emails (which is not your fault,[ the regex can get quite complicated](https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression).

Perhaps just leave it to the browser to do the [validation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#validation) & remove the pattern. Note that this will require you to make this a form (will also allow the user to submit the form by just pressing enter/the return key on their phone, which is quite desirable)

Reply:
Don't really think it matters that it is not perfect because it will still filter out the most obvious errors. I've removed it though because the backend will return "Invalid login credentials." if the email is wrong, which is sufficient I think.